src/plugins/intel_gpu/src/graph/space_to_depth.cpp

```diff
@@ -2,16 +2,55 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#include "openvino/op/space_to_depth.hpp"
 #include "space_to_depth_inst.h"
 
 #include "primitive_type_base.h"
 #include "intel_gpu/runtime/error_handler.hpp"
 #include "json_object.h"
+#include "data_inst.h"
 #include <string>
 
+#include "space_to_depth_shape_inference.hpp"
+
 namespace cldnn {
 GPU_DEFINE_PRIMITIVE_TYPE_ID(space_to_depth)
 
+template<typename ShapeType>
+std::vector<layout> space_to_depth_inst::calc_output_layouts(space_to_depth_node const& node, kernel_impl_params const& impl_param) {
+    auto desc = impl_param.typed_desc<space_to_depth>();
+    auto input_layout = impl_param.get_input_layout(0);
+
+    auto output_type = desc->output_data_types[0].value_or(input_layout.data_type);
+    if (impl_param.has_fused_primitives())
+        output_type = impl_param.get_fused_output_layout().data_type;
+
+    auto output_format = input_layout.format;
+
+    ov::op::v0::SpaceToDepth op;
+    op.set_block_size(desc->block_size);
+    switch (desc->mode) {
+    case space_to_depth::depth_mode::depth_first:
```

Comment:
Could we fix the primitive to use ov's mode definition itself, instead of translating here?

Reply:
Modified to use ov's mode definition. other all reference of depth_mode of space_to_depth use ov's mode.